internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -701,40 +735,50 @@ func (r *ClusterDeploymentsReconciler) SetupWithManager(mgr ctrl.Manager) error
 		return reply
 	}
 
+	// Reconcile the ClusterDeployment referenced by this AgentClusterInstall.
+	mapClusterInstallToClusterDeployment := func(a client.Object) []reconcile.Request {
+		aci, ok := a.(*hiveext.AgentClusterInstall)
+		if !ok {
+			r.Log.Errorf("%v was not an AgentClusterInstall", a) // shouldn't be possible
```

Comment:
it's possible if there will be other installers.

Reply:
Line 759 where this is used is watching only AgentClusterInstall, so this should not be called for any other type. Type check just a formality in case someone does something weird with it later.